dev/generate_bom.py

```diff
@@ -110,6 +110,8 @@ def init_argument_parser(cls, parser):
                         help="GCE project we publish HA Spinnaker component images to.")
     parser.add_argument('--git_prefix', default='https://github.com/spinnaker',
                         help="Prefix to the component source URIs.")
+    parser.add_argument('--maven_override', default='http://spinnaker.bintray.com/spinnaker',
```

Comment:
This is a confusing name. You have an override file, but htis is specifying a repository. I'd call this something like -maven_override_repository. However it isnt clear if this is the default repo or not, in which case it isnt --maven_override_repository, rather --maven_repository. However  it is only spinnaker, so maybe this is the --spinnaker_maven_repository.

It isnt clear how this goes with the maven override. It seems simpler if you always have the override file (in which case it is no longer an override file) but that requires scripting. So this is looking too messy and complicated. I dont understand it well enough to provide an alternative. We can talk offline.